src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -269,6 +269,8 @@ private void UpdateReferenceLocationsTask(Task<IInlineRenameLocationSet> allRena
                 async t =>
                 {
                     await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationTokenSource.Token);
```

Comment:
Why doesn't `SwitchToMainThreadAsync` itself throw OperationCancelledException if cancellation was requested?

Reply:
`SwitchToMainThreadAsync` throws only if the switch to the main thread fails. After cancellation is requested, the continuation is allowed to execute on either a background thread or the main thread. If the main thread ends up running the continuation, the switch is considered successful and there is no need to throw.

Reply:
@sharwell Where was this stated in the contract behavior? It's definitely in the implementation, but it was oddly absent from any docs I could find.

Reply:
@jasonmalinowski It's here:
https://github.com/Microsoft/vs-threading/blob/fd039889435ba4558dea4d309d9f956567355555/src/Microsoft.VisualStudio.Threading/JoinableTaskFactory.cs#L149-L150

Reply:
The wording could be a bit clearer, but I wouldn't expect the behavior to change.

Reply:
I guess it depends on your reading of "when <paramref name="cancellationToken" /> is canceled before any required transition to the main thread is complete.". In this case, the transition to the main thread *did* happen, so as far as I'm concerned, that statement doesn't apply. Behavior is unspecified.

More than anything else, I think this means yes it needs to be reworded. Even if it's 100% correct, this is a good example where precise speech might be useless if it's not clear.

Reply:
Drawing an analogy here, consider cancelling an order on Amazon. Amazon will say "if you cancel your order before we start asynchronously shipping your order, we will refund you." That says nothing about what happens if the order comes when the order is already being processed. They could refund me and tell the shipper to return the package, or just tell me to accept my order.

Reply:
And oh, I'll admit I'm being very lawyery here and a bit obtuse...on purpose. So bear with me. :smile: